packages/frontend-core/src/components/grid/cells/SignatureCell.svelte

```diff
@@ -0,0 +1,185 @@
+<script>
+  import { onMount, getContext } from "svelte"
+  import { SignatureModal } from "@budibase/frontend-core/src/components"
+  import { CoreSignature, ActionButton } from "@budibase/bbui"
+
+  export let schema
+  export let value
+  export let focused = false
+  export let onChange
+  export let readonly = false
+  export let api
+  export let invertX = false
+  export let invertY = false
+
+  const { API, notifications, props } = getContext("grid")
+
+  let isOpen = false
+  let editing = false
+  let signature
+  let modal
+
+  $: if (value) {
+    const [attachment] = value
+    signature = attachment
+  }
+
+  $: editable = focused && !readonly
+  $: {
+    if (!focused) {
+      close()
+    }
+  }
+
+  const onKeyDown = () => {
+    return isOpen
```

Comment:
If the signature component doesn't actually support any keyboard events (which it looks like it doesn't), then it would be better to return `false` always here - it means that using keys like tab or escape will always work to close the cell.